api/src/services/dataset.js

```diff
@@ -495,31 +520,23 @@ async function add_files({ dataset_id, data }) {
  * txA: create -> dataset created
  * txB: create -> unique constraint violation
  *
+ * @param {Object} data - The data object containing details of the dataset to be created.
  * @returns {Promise<Object|undefined>} The created dataset object or undefined if a dataset with the same name and type already exists.
  */
 function createDataset(data) {
-  return prisma.$transaction(async (tx) => {
-    // find if a dataset with the same name and type already exists
-    const existingDataset = await tx.dataset.findFirst({
-      where: {
-        name: data.name,
-        type: data.type,
-        is_deleted: false,
-      },
-      select: {
-        id: true,
-      },
-    });
-    if (existingDataset) {
-      return;
-    }
-    // if it doesn't exist, create it
-    return tx.dataset.create({
-      data,
-    });
-  });
+  return prisma.$transaction(async (tx) => createDatasetInTransaction(tx, data));
```

Comment:
Extracted this functionality into `createDatasetInTransaction` for re-usability.